aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -1307,14 +1271,40 @@ void TensorIterator::build(TensorIteratorConfig& config) {
   view_offsets_ = DimVector(ndim_offsets, 0);
 }
 
-SplitUntil32Bit TensorIterator::with_32bit_indexing() const {
+void TensorIterator::set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) {
+  auto& op = operands_[output_idx];
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(output_idx < num_outputs_);
+  if (!op.tensor.defined()) {
+      TORCH_INTERNAL_ASSERT(op.is_type_defined(), "no type for operand", output_idx);
```

Comment:
On a quick read it looks like a less aggressive factoring would avoid some redundant tests when this is called from `allocate_or_resize_outputs`, at the cost of leaving more code in `fast_set_up` - but I'm assuming you figure the slow path can afford it (or that there's some API requirement that it be done in `set_output` that I'm missing)

Reply:
This is a good question. When I wrote this to start, I tried to assume as few preconditions on the call-site as possible, ergo the redundant tests. If I understand you correctly, it's just this assert that could be removed (I don't think any of the other asserts are removable). Seems reasonable.